java/google/registry/rde/imports/RdeContactImportAction.java

```diff
@@ -120,8 +126,33 @@ private RdeImportUtils createRdeImportUtils() {
     }
 
     @Override
-    public void map(ContactResource contact) {
-      getImportUtils().importContact(contact);
+    public void map(JaxbFragment<XjcRdeContactElement> fragment) {
+      final XjcRdeContact xjccontact = fragment.getInstance().getValue();
+      try {
+        logger.infofmt("Converting xml for contact %s", xjccontact.getId());
+        // Record number of attempted map operations
+        getContext().incrementCounter("contact-import-attempted");
+        ofy().transact(new VoidWork() {
+          @Override
+          public void vrun() {
+            ContactResource contact =
+                XjcToContactResourceConverter.convertContact(xjccontact);
+            logger.infofmt("Saving contact %s", contact.getContactId());
+            getImportUtils().importContact(contact);
+            logger.infofmt("Contact %s was imported successfully", contact.getContactId());
+          }
+        });
+        getContext().incrementCounter("contact-import-saved");
+      } catch (ResourceExistsException e) {
```

Comment:
Did you consider handling this exception inside the transaction?  I'm trying to decide if that's better, and I'm not sure.